src/syft/lib/tensor/tensorbase.py

```diff
@@ -0,0 +1,172 @@
+# stdlib
+from functools import partial
+from typing import Any
+from typing import List
+from typing import Union
+
+# third party
+import numpy as np
+import torch
+
+# syft absolute
+from syft.decorators import syft_decorator
+from syft.lib.tensor.tensorbase_util import call_func_and_wrap_result
+
+Num = Union[int, float]
+
+
+class ChildDelegatorTensor:
+    def __getattr__(self, name: str) -> Any:
+        func_or_attr = getattr(self.child, name)
+        if callable(func_or_attr):
+            return partial(call_func_and_wrap_result, func_or_attr, self.__class__)
+        else:
+            var = func_or_attr
+            return var
+
+
+class DataTensor(ChildDelegatorTensor):
+    @syft_decorator(typechecking=True)
```

Comment:
Q: @tudorcebere we should remove the ```syft_decorator```, right?

Reply:
Don't forget about this :D

Reply:
I removed all of these. 😊